tortoise/models.py

```diff
@@ -598,35 +580,74 @@ def __search_for_field_attributes(base: Type, attrs: dict) -> None:
                                 field_name="pk", field=field, source_field=source_field
                             )
                         )
-
-        # Clean the class attributes
-        for slot in fields_map:
-            attrs.pop(slot, None)
-        attrs["_meta"] = meta = MetaInfo(meta_class)
-
+        return (fields_map, filters, fk_fields, m2m_fields, o2o_fields)
+
+    @staticmethod
+    def build_meta(
+        meta_class,
+        fields_map,
+        fields_db_projection,
+        filters,
+        fk_fields,
+        o2o_fields,
+        m2m_fields,
+        pk_attr,
+    ) -> MetaInfo:
+        meta = MetaInfo(meta_class)
         meta.fields_map = fields_map
         meta.fields_db_projection = fields_db_projection
         meta._filters = filters
         meta.fk_fields = fk_fields
-        meta.backward_fk_fields = set()
         meta.o2o_fields = o2o_fields
-        meta.backward_o2o_fields = set()
         meta.m2m_fields = m2m_fields
-        meta.default_connection = None
         meta.pk_attr = pk_attr
-        meta.pk = fields_map.get(pk_attr)  # type: ignore
-        if meta.pk:
-            if meta.pk.source_field:
-                meta.db_pk_column = meta.pk.source_field
-            elif isinstance(meta.pk, OneToOneFieldInstance):
-                meta.db_pk_column = f"{meta.pk_attr}_id"
+        if pk_field := fields_map.get(pk_attr):
+            meta.pk = pk_field  # type:ignore
+            if pk_field.source_field:
+                meta.db_pk_column = pk_field.source_field
+            elif isinstance(pk_field, OneToOneFieldInstance):
+                meta.db_pk_column = f"{pk_attr}_id"
             else:
-                meta.db_pk_column = meta.pk_attr
+                meta.db_pk_column = pk_attr
         meta._inited = False
         if not fields_map:
             meta.abstract = True
+        return meta
+
+    def __new__(cls, name: str, bases: tuple[Type, ...], attrs: dict[str, Any]) -> "ModelMeta":
```

Comment:
`__new__` is the main method of `ModelMeta` and it invokes `build_meta` and `parse_custom_pk`. I think it makes more sense to have it above `build_meta` and `parse_custom_pk` in the code because the person reading the code would want to start reading the code from the entry point.